doc/source/ray-core/objects/serialization.rst

```diff
@@ -23,6 +23,23 @@ Plasma is used to efficiently transfer objects across different processes and di
 
 Each node has its own object store. When data is put into the object store, it does not get automatically broadcasted to other nodes. Data remains local to the writer until requested by another task or actor on another node.
 
+Object Refs
+~~~~~~~~~~~~~~~~~~~
+
+Ray object refs can be serialized using `ray.cloudpickle`. The object ref can then be deserialized and accessed with `ray.get()`. Note that `ray.cloudpickle` must be used; other pickle tools are not guaranteed to work. Additionally, the process that deserializes the `ObjectRef` must be part of the same Ray cluster that serialized it.
+
+When serialized, the `ObjectRef`'s value will remain pinned in Ray's shared memory object store. The object must be explicitly freed by calling `ray._private.internal_api.free(obj_ref)`.
+
+This should be used as a last resort. Passing `ObjectRefs` through Ray task arguments and return values is the recommended approach.
+
+.. warning::
+  
+  `ray._private.internal_api.free(obj_ref)` is a private API and may be changed in future Ray versions.
+
+This code example demonstrates how to serialize an object ref, store it in external storage, deserialize and use it, and lastly free its object.
+
+.. literalinclude:: /ray-core/doc_code/object_ref_serialization.py
```

Comment:
Can you make sure this code runs as part of the doctests? There are some examples in the other core docs pages that you can follow to enable this.

Reply:
I assumed adding the file to the `doc_code` directory makes it automatically run as part of the doctests. I don't see any special handling for other code files in the `doc_code` directory. Is there something I need to explicitly enable?

Reply:
Hmm I think you need the `testcode` directive in rst. @jjyao is that right?

Reply:
I think `testcode` is for code that written directly in the `rst` file rather than code that's imported from `doc_code`. But that's just a guess– I'll defer to @jjyao.

Reply:
`literalinclude` code is automatically tested.

In your case, the code snippet is short enough, you can consider have it in place in rst with `testcode`.